annotation/annotation/annotations/main.py

```diff
@@ -110,8 +113,12 @@ def connect_s3(bucket_name: str) -> boto3.resource:
         # it responses with TypeError: NoneType object is not
         # callable, this try/except block should be changed after understanding
         # what is going on
-    except TypeError:
-        raise s3_resource.meta.client.exceptions.NoSuchBucket
+
+        # It throws S3 Client Error 404 now
+    except ClientError as e:
+        raise s3_resource.meta.client.exceptions.NoSuchBucket(
+            e.response, e.operation_name
+        )
     return s3_resource
```

Comment:
Could you chain NoSuchBucket exception from caught error to preserve a traceback?